pkg/daemon/daemon.go

```diff
@@ -101,10 +104,24 @@ func NewDaemon() (Daemon, error) {
 
 func (d *daemon) Run() {
 	glog.Info("daemon Run():")
-	go wait.Forever(d.AddPeriodicUpdate, time.Duration(d.config.PeriodicUpdate)*time.Second)
-	go wait.Forever(d.DeletePeriodicUpdate, time.Duration(d.config.PeriodicUpdate)*time.Second)
+	// closing the channel will stop the goroutines executed in the wait.Until() calls below
+	stopPeriodicsChan := make(chan struct{})
+	// Run periodic tasks
+	go wait.Until(d.AddPeriodicUpdate, time.Duration(d.config.PeriodicUpdate)*time.Second, stopPeriodicsChan)
+	go wait.Until(d.DeletePeriodicUpdate, time.Duration(d.config.PeriodicUpdate)*time.Second, stopPeriodicsChan)
 	glog.Info("watcher Run():")
-	d.watcher.Run()
+	// Running Watcher
+	// calling watcherStopFunc() will stop the watcher which listens to events on a separate goroutine
+	watcherStopFunc := d.watcher.Run()
```

Comment:
you are hiding the go routines in the watcher.run which is confusing even with the comment.  Why not go d.watcher.Run()?
also can we use one channel to close all go routines by passing it in the d.watcher.Run?  (and remove the watcherStopFunc)

Reply:
ok let do this, defer watcherStopFunc()

Reply:
so let change Run to RunInBackground